app/main.py

```diff
@@ -1,2 +1,65 @@
+from __future__ import annotations
+from copy import deepcopy
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [[None, None] for _ in range(8)]
+        self.capacity = 8
+
+    def key_upp(self, key: int) -> int:
+        key += 1
+        if key == self.capacity:
+            key = 0
+        return key
+
+    def write(self, index: int, key: Any, value: Any) -> None:
+        self.hash_table[index][0] = key
+        self.hash_table[index][1] = value
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        old_hash_tabel = deepcopy(self.hash_table)
+        self.hash_table = [[None, None] for _ in range(self.capacity)]
+        for pair_key_value in old_hash_tabel:
+            if pair_key_value[0] is None:
+                continue
+            index = hash(pair_key_value[0]) % self.capacity
+            for _ in range(self.capacity):
+                if self.hash_table[index][0] is None:
+                    self.write(index, pair_key_value[0], pair_key_value[1])
+                    break
+                index += 1
+                if index == self.capacity:
+                    index = 0
```

Comment:
U can use __setitem__ method instead of this block of code
also u don't change the len of dict here